services/core-web/src/components/noticeOfWork/NOWActionWrapper.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+import { withRouter } from "react-router-dom";
+import CustomPropTypes from "@/customPropTypes";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+import { getNoticeOfWork } from "@common/selectors/noticeOfWorkSelectors";
+
+/**
+ * @constant NOWActionWrapper conditionally renders NoW actions based on various conditions (ie, Rejected, Permit issued, client delay, stages not started, etc)
+ * persists permissions using authWrapper - These actions are not visible to admin if disabled.
+ */
+
+const propTypes = {
+  noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
+  history: PropTypes.shape({ location: PropTypes.objectOf(PropTypes.any) }).isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.element.isRequired),
+    PropTypes.element.isRequired,
+  ]).isRequired,
+};
+
+const defaultProps = {};
+// eslint-disable-next-line react/prefer-stateless-function
+export class NOWActionWrapper extends Component {
+  // state = { currentTab: "" };
+
+  // componentDidMount() {
+  //   // use this logic to disable buttons on a specific section based on progress not started or completed
+  //   const split = this.props.history.location.pathname.split("/");
+  //   const currentTab = split.pop();
+  //   this.setState({ currentTab });
+  // }
+
+  render() {
+    const disabled = false;
+    return !disabled ? (
+      <AuthorizationWrapper {...this.props}>
+        {React.createElement("span", null, this.props.children)}
+      </AuthorizationWrapper>
+    ) : (
+      <div />
+    );
+  }
+}
+
+NOWActionWrapper.propTypes = propTypes;
+NOWActionWrapper.defaultProps = defaultProps;
+
+const mapStateToProps = (state) => ({
+  // can disable all actions based off application status === rejected || withdrawn
```

Comment:
Or, are you thinking we map the conditional here based on multiple gets?

Reply:
Yeah, I assumed that all logic would be stored in redux, depending on where the "complete" states are saved, it might not be the noticeofWork object - but I tested the logic with the current now_status and worked well!